service/history/hsm/tree.go

```diff
@@ -386,18 +408,19 @@ func (n *Node) DeleteChild(key Key) error {
 		return err
 	}
 
-	// Mark entire subtree as deleted
+	// Mark entire subtree as deleted and record all deletions
+	root := n.root()
 	if err := child.Walk(func(n *Node) error {
 		n.cache.deleted = true
+		// Record deletion for every node in subtree
```

Comment:
No need, these will be compacted anyways.

Reply:
Indeed! Thx